### PR TITLE
Add leaderboard avatars and badges

### PR DESCRIPTION
**Pull request by Bruception**
### Description

- Adds discord avatar and badges to leaderboard entry
- Updated leaderboard entry for leaderboards to be more consistent with daily entry
- Updated documentation

### CODE REVIEW

File: backend/src/api/controllers/leaderboard.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export async function getLeaderboard(
   const normalizedLeaderboard = _.map(leaderboard, (entry) => {
     return uid && entry.uid === uid
       ? entry
-      : _.omit(entry, ["discordId", "uid", "difficulty", "language"]);
+      : _.omit(entry, ["_id", "difficulty", "language"]);
```
Comment (Bruception) [inline backend/src/api/controllers/leaderboard.ts:43]: We need discord id to display avatars

**Comment by Bruception** [comment]
Need to rework this to make badges more extensible

File: backend/src/api/controllers/leaderboard.ts (hunk 1 above)
Comment (Ferotiq) [reply backend/src/api/controllers/leaderboard.ts:43]: are uids okay to expose publicly too?

Comment (Bruception) [reply backend/src/api/controllers/leaderboard.ts:43]: Yes

**Review by Miodec — APPROVED** [review]
Thanks

**Pull request merged**